### ISSUE TITLE
Curl on some systems does not support --json

### ISSUE THREAD

**Issue by EricH9958** [OliveTin/OliveTin#462]
In Ubuntu 22.04 trying to set up basic user auth. 

From the online docs:

authLocalUsers:
  enabled: true
  users:
    - username: james
       password: $argon2id$v=19$m=65536,t=4,p=6$LnNW4sw+jZfa5Ex3YjfuHQ$vl8pjUJhxNmBxScV4lI3cgAZPkNB1rSrnX6ibgoAP8k

To set up the password it shows:

curl -sS --json '{"password": "myPassword"}' http://olivetin.example.com:1337/api/PasswordHash

returned this error

root@Server-2:~# curl -sS --json '{"password": "XXXXXXXXXXXXXX"}' http://192.168.1.102:1337/api/PasswordHash
curl: option --json: is unknown
curl: try 'curl --help' or 'curl --manual' for more information

I was able to complete with this code:

The error you’re seeing, curl: option --json: is unknown, occurs because the --json option isn’t available in standard curl versions.

To send JSON data in a curl request, you need to use the -d option (for data) along with the -H option to set the content type to application/json. Here’s the corrected command:

curl -sS -H "Content-Type: application/json" -d '{"password": "XXXXXXXXXXX"}' http://192.168.1.102:1337/api/PasswordHash

This returned:

oot@Server-2:~# curl -sS -H "Content-Type: application/json" -d '{"password": "XXXXXXXXX"}' http://192.168.1.102:1337/api/PasswordHash
Your password hash is: $argon2id$v=19$m=65536,t=4,p=8$e7dGmV+XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX+sw

**Comment by jamesread** [comment]
Hey @EricH9958 , oh, weird, it sounds like Ubuntu's curl does not support --json (I use Fedora Linux, which has a curl that supports --json).

You've got a pretty old version of Ubuntu there, I wonder if more recent releases support it.

What is your `curl --version` please?

**Comment by jamesread** [comment]
Ubuntu 24.04 has curl 8.6 which supports --json. 

I'll probably patch the docs to include your example for users with older version of curl though. Just curious which version you have in Ubuntu 22.04.

**Comment by EricH9958** [comment]
Hi James, 

I'm running Ubuntu 22.04.1  Since it runs my Plex server and everything is running well I'm not going to upgrade to 24.04 for a while.  As far as I can tell there isn't really any upside to upgrading for now. 


root@Server-2:~# curl --version
curl 7.81.0 (x86_64-pc-linux-gnu) libcurl/7.81.0 OpenSSL/3.0.2 zlib/1.2.11 brotli/1.0.9 zstd/1.4.8 libidn2/2.3.2 libpsl/0.21.0 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.43.0 librtmp/2.3 OpenLDAP/2.5.18
Release-Date: 2022-01-05
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL TLS-SRP UnixSockets zstd

**Comment by jamesread** [comment]
Hey @EricH9958 , ah yeah, it looks like it was added in 7.82 - just one minor release away from where you are (source: https://curl.se/docs/manpage.html). 

I think that given that it's in modern versions of curl, I'm going to close this as not-a-bug, and as people update/upgrade it won't be an issue. However, I have updated the local-users doc page to point to this issue when we use the --json option, so if other people have this problem they can use your suggested solution. I hope that's OK. :-) 

![image](https://github.com/user-attachments/assets/23981421-08f2-4f6d-9ede-1bc1b7e9390e)

Source: https://docs.olivetin.app/local-users.html
